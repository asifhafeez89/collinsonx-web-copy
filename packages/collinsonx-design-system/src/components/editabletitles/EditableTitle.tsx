import Head from 'next/head';
import Heading from '../heading/Heading';
import { Box, Flex, Stack } from '@mantine/core';
import { Pencil } from '../../assets/icons';
import Link from 'next/link';
import colors from '../../colour-constants';

interface EditableTitlesProps {
  title: string;
  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'legend';
  children: React.ReactNode;
  to?: string | null;
  showBorder?: boolean;
}

export default function EditableTitle({
  title,
  as,
  children,
  to = null,
  showBorder = true,
}: EditableTitlesProps) {
  return (
    <Stack
      p={{ base: '20px', xl: '20px 0' }}
      sx={{
        borderBottom: showBorder
          ? `1px solid ${colors.borderSection}`
          : 'initial',
        paddingBottom: '20px',

        '@media (max-width: 768px)': {
          backgroundColor: '#fff',
          width: '100%',
          borderBottom: 'initial',
          paddingBottom: '2rem',
        },
      }}
    >
      <Flex direction="row" justify="space-between" align="center">
        <Heading as={as} margin={0} padding={0}>
          {title}
        </Heading>

        {to !== null && (
          <Link href={to}>
            {' '}
            <Pencil />
          </Link>
        )}
      </Flex>
      {children}
    </Stack>
  );
}
