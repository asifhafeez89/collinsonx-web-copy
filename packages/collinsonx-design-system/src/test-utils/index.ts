import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
