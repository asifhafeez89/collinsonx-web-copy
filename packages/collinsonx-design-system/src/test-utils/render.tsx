import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
