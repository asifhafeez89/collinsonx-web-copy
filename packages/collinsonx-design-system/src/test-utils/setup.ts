import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
