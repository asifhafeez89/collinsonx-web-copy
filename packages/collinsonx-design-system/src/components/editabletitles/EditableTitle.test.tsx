import renderer from 'react-test-renderer';
import EditableTitle from './EditableTitle';
import { Provider } from 'test-utils';

describe('<EditableField />', () => {
  it('renders Details', () => {
    const tree = renderer
      .create(
        <Provider>
          <EditableTitle title="This is a title" as="h2" to="">
            <div>Test</div>
          </EditableTitle>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
