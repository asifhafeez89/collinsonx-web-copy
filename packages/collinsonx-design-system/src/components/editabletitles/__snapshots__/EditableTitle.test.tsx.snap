// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditableField /> renders Details 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 1rem;
  border-bottom: 1px solid #b4b4b4;
  padding-bottom: 20px;
  padding: 20px;
}

@media (max-width: 768px) {
  .emotion-1 {
    background-color: #fff;
    width: 100%;
    border-bottom: initial;
    padding-bottom: 2rem;
  }
}

@media (min-width: 88em) {
  .emotion-1 {
    padding: 20px 0;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3 {
  position: relative;
  display: block;
  color: #000000;
}

<div
  className="emotion-0 emotion-1"
>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-3"
      color="#000000"
    >
      <h2
        style={
          {
            "margin": 0,
            "padding": 0,
          }
        }
      >
        This is a title
      </h2>
    </div>
    <a
      href=""
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
       
      <test-file-stub />
    </a>
  </div>
  <div>
    Test
  </div>
</div>
`;
